source/postprocess/dynamic_topography.cc

```diff
@@ -110,9 +115,14 @@ namespace aspect
               // for each of the quadrature points, evaluate the
               // stress and compute the component in direction of the
               // gravity vector
+              
+              double dynamic_topography_x_volume = 0;
+              double volume = 0;
+              Point<dim> location = fe_values.quadrature_point(0);
+
               for (unsigned int q=0; q<quadrature_formula.size(); ++q)
```

Comment:
The quadrature has only one quadrature point, right? So you could skip the loop like you do for the surface.